lib/OpenQA/WebAPI/Controller/API/V1/Job.pm

```diff
@@ -446,8 +446,13 @@ sub create_artefact {
         $self->render_later;    # XXX: Not really needed, but in case of upstream changes
         my @ioloop_evs = qw(events);
         my @evs        = @{Mojo::IOLoop->singleton}{@ioloop_evs};
+
+        # See: https://mojolicious.org/perldoc/Mojolicious/Guides/FAQ#What-does-Connection-already-closed-mean
+        my $tx = $self->tx;     # NOTE: Keep tx around as long operations could make it disappear
+
         return Mojo::IOLoop->subprocess(
             sub {
+                $tx;
```

Comment:
This is just to make sure that $tx ends up in the coderef context right? , loooks like something is missing

Reply:
Yes, See: https://mojolicious.org/perldoc/Mojolicious/Guides/FAQ#What-does-Connection-already-closed-mean

Reply:
Yea, the controller object (`Mojolicious::Controller`) only keeps a weaked reference to the transaction (`$c->tx`). So if you want to use code depending on the transaction (like `$c->param(...)`) in async closures, you need to keep a strong reference to the transaction somewhere.

Reply:
From a framework design point of view this is a very tricky problem. If `$c->tx` wasn't a weakened reference you wouldn't have this problem here, but pretty much every Mojolicious application using async features would be required to use `weaken` all over the place or leak tons of memory with every request. So, i think this is actually the lesser evil. 😩

Reply:
@kraih Yeah, I guess so... having memory leaks all over the place sounds like a fun thing to debug :)